images/storage-network-discoverer/src/main.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	goruntime "runtime"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"storage-network-controller/internal/config"
+	"storage-network-controller/internal/logger"
+	"storage-network-controller/pkg/discoverer"
+)
+
+func KubernetesDefaultConfigCreate() (*rest.Config, error) {
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	// Get a config to talk to API server
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("config kubernetes error %w", err)
+	}
+	return config, nil
+}
+
+// have a separate function so we can return an exit code w/o skipping defers
+func runController(ctx context.Context, cfg *config.Options, log *logger.Logger) int {
```

Comment:
Looking at the code again I like the idea having exit code a part of the `main` routine. I think it will be clear to return error here.

Reply:
Actually they [have a module](https://pkg.go.dev/github.com/square/exit) for exit code type.

Reply:
Don't think we need it. Returning error should be fine here

Reply:
Nice tip! Thank you :-) I did as you suggest (with 'exit' module)